### ISSUE TITLE
BUG: qcut fails with Float64Dtype

### ISSUE THREAD

**Issue by tamargrey** [pandas-dev/pandas#40730]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
series = pd.Series([1.0, 2.0, 3.0, 4.4], dtype=pd.Float64Dtype())
pd.qcut(series, 2)
```

#### Problem description
`pd.qcut` currently accepts the nullable `Int64Dtype` as well as `'float64'`, so I would expect it to work with the `Float64Dtype` as well. Instead the following error is produced:

```python-traceback
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-29-1db98f70db38> in <module>
      1 series = pd.Series([1.0,2.0,3.0,4.0], dtype=pd.Float64Dtype())
----> 2 pd.qcut(series, 2)

~/.pyenv/versions/3.8.2/envs/woodwork/lib/python3.8/site-packages/pandas/core/reshape/tile.py in qcut(x, q, labels, retbins, precision, duplicates)
    356         quantiles = q
    357     bins = algos.quantile(x, quantiles)
--> 358     fac, bins = _bins_to_cuts(
    359         x,
    360         bins,

~/.pyenv/versions/3.8.2/envs/woodwork/lib/python3.8/site-packages/pandas/core/reshape/tile.py in _bins_to_cuts(x, bins, right, labels, precision, include_lowest, dtype, duplicates, ordered)
    408 
    409     if include_lowest:
--> 410         ids[x == bins[0]] = 1
    411 
    412     na_mask = isna(x) | (ids == len(bins)) | (ids == 0)

IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
```

#### Expected Output
Should match that of `float64`

```
0    (0.999, 2.5]
1    (0.999, 2.5]
2      (2.5, 4.0]
3      (2.5, 4.0]
dtype: category
Categories (2, interval[float64]): [(0.999, 2.5] < (2.5, 4.0]]
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : f2c8480af2f25efdbd803218b9d87980f416563e
python           : 3.8.2.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.6.0
Version          : Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.3
numpy            : 1.19.5
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 41.2.0
Cython           : None
pytest           : 6.0.1
hypothesis       : None
sphinx           : 3.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.18.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : 0.8.7
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 3.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

**Comment by jorisvandenbossche** [comment]
@tamargrey Thanks for the report!

This was fixed last year for the nullable Int64Dtype in https://github.com/pandas-dev/pandas/pull/31440, but it seems we need to generalize the fix to also cover Float64Dtype. 

In this location:

https://github.com/pandas-dev/pandas/blob/5c64037034deda69d7af6d676d0d70f9ff359e4e/pandas/core/reshape/tile.py#L482-L487

we need to change the `is_integer_dtype` to generally catch nullable numeric dtypes

**Comment by lithomas1** [comment]
Fixed by #40969
